src/backend/sev/unattested_launch.rs

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! The `unattested_launch` is an implementation of a so-called "synthetic"
+//! or "fake" client that speaks the remote AMD SEV launch protocol. This
+//! exists for several reasons:
+//!
+//!   1.) There is no special-casing in the SEV backend launch process. It
+//!       will speak the same protocol for all launch use-cases. The
+//!       synthetic client allows us to avoid special-casing any of the launch
+//!       code.
+//!   2.) The synthetic client allows for keeps to be launched in solitary
+//!       environments such as CI or other testing where there are no actual
+//!       clients who will participate in the launch protocol.
+//!   3.) Sometimes you just want a keep and you don't care to attest.
+//!   4.) The synthetic client, as a side-effect, also acts as some degree
+//!       of test coverage for the client-side of the remote attestation
+//!       protocol.
+
+use std::convert::TryFrom;
+use std::os::unix::net::UnixStream;
+
+use koine::attestation::sev::*;
+use serde::de::Deserialize;
+use serde_cbor as serde_flavor;
+use sev::launch::Policy;
+use sev::session::Session;
+
+pub fn launch(sock: UnixStream) {
+    let mut de = serde_flavor::Deserializer::from_reader(&sock);
+    let chain_packet =
+        Message::deserialize(&mut de).expect("failed to deserialize expected certificate chain");
+    assert!(matches!(
+        chain_packet.mimetype,
+        MimeType::CertificateChainNaples | MimeType::CertificateChainRome
+    ));
+    let chain = serde_flavor::from_reader(chain_packet.payload.as_slice())
+        .expect("failed to deserialize certificate chain");
+
+    let policy = Policy::default();
+    let session = Session::try_from(policy).expect("failed to craft policy");
+
+    let start = session.start(chain).expect("failed to start session");
+    let mut start_packet = Message {
+        mimetype: MimeType::LaunchStart,
+        payload: vec![],
+    };
+    serde_flavor::to_writer(&mut start_packet.payload, &start)
+        .expect("failed to serialize launch start");
+    serde_flavor::to_writer(&sock, &start_packet).expect("failed to send launch start");
+
+    // Discard the measurement, the synthetic client doesn't care
+    // for an unattested launch.
+    let msr =
+        Message::deserialize(&mut de).expect("failed to deserialize expected measurement packet");
+    assert_eq!(msr.mimetype, MimeType::Measurement);
+
+    let secret_packet = Message {
+        mimetype: MimeType::Secret,
+        payload: vec![],
```

Comment:
We might want to put in a well known secret, so we can recognize it in the payload or shim.

Reply:
I'd like to follow up on this in a separate PR.